src/components/camera/client.hpp

```diff
@@ -19,8 +20,10 @@ class CameraClient : public Camera {
    protected:
     CameraClient(std::string name);
     std::unique_ptr<viam::component::camera::v1::CameraService::StubInterface> stub_;
-    std::string name_;
 
    private:
     std::shared_ptr<grpc::Channel> channel_;
 };
+
+template std::shared_ptr<CameraClient> typed_resource_from_robot(const std::shared_ptr<RobotClient>,
```

Comment:
I don't think you need this explicit instantiation. Though, there might be a reason to do it. I'd expect to see things organized a little differently though if that was what you were after (in particular, I wouldn't expect to see a definition of typed_resource_from_robot in a header).